pkg/reconciler/common/read_releases.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"golang.org/x/mod/semver"
+)
+
+const (
+	KoEnvKey = "KO_DATA_PATH"
+)
+
+// RetrieveManifestPath returns the manifest path for Knative based a provided version and component
+func RetrieveManifestPath(version, kcomponent string) string {
+	koDataDir := os.Getenv(KoEnvKey)
+	return filepath.Join(koDataDir, kcomponent, version)
+}
+
+func isFirstVersionMoreRecent(former, latter string) bool {
+	return semver.Compare(addMissingLetterV(former), addMissingLetterV(latter)) == 1
+}
+
+func addMissingLetterV(version string) string {
```

Comment:
Also: This is only used inside the compare function which is only used when listing releases. Why don't we just always create directories with correct semver versions and drop this entirely?

Reply:
@markusthoemmes That works for me.
However, we need to convert the version anyhow, coz we use the version without `v` to set the status version. We still need to convert the version.
To get rid of the version conversion, we need to make sure spec.version and status.version also include `v` at the beginning. Do we have to enforce users to do that?

Reply:
Then let's consistently **not** use a `"v"` and always add it. Once we start dealing with user input, we can think about sanitizing just that but I wouldn't want to sanitize everything.

Reply:
Then let's consistently **not** use a `"v"` and always add it. Once we start dealing with user input, we can think about sanitizing just that but I wouldn't want to sanitize everything.

Reply:
That's what I doing now. Keep x.y.z without `v`, and use it everywhere. Once we need to compare the versions, add `v` in front.

Reply:
It's not quite though. You're always either adding or not adding the version. Do we want users to be able to put `v0.15` and `0.15` in there? I kinda think we only want one of those to work honestly.

Reply:
If we decide to use 0.15.x all the time without `v`, how could that change the this PR? I am confused how that will affect the code here?

Reply:
All the directories are named after the version without `v`, so `addMissingLetterV` can add it before comparison.
spec.version and status.version are without `v`, so `addMissingLetterV` can add it before comparison.